### ISSUE TITLE
gpu.setActiveBuffer crashes OCEmu

### ISSUE THREAD

**Issue by Xandaros** [zenith391/OCEmu#35]
Wanting to try out the new and shine VRAM buffers, I quickly discovered that trying to use them will crash OCEmu.

I tried using this script:
```lua
local component = require("component")
local gpu = component.gpu

local buffer = gpu.allocateBuffer()
gpu.setActiveBuffer(buffer)
gpu.freeBuffer()
```
And running it will immediately close OCEmu with this output:
```
false	bad arguments #1 (number expected, got nil)
stack traceback:
	[C]: in function 'error'
	./main.lua:526: in function 'resume_thread'
	./main.lua:568: in function '?'
	boot.lua:310: in function <boot.lua:77>
	[C]: in function 'xpcall'
	boot.lua:327: in main chunk
	[C]: in ?
```

Trying the same script in-game does exactly what I expected - nothing. :D

**Comment by zenith391** [comment]
It's actually a problem both in your code and in OCEmu:
You should do `gpu.freeBuffer(buffer)` instead of `gpu.freeBuffer()`
And in OCEmu, I forgot to handle the case where you pass no argument to freeBuffer.

**Comment by Xandaros** [comment]
Thank you for the quick response!

However, I have to disagree. According to the docs:
`freeBuffer([index: number]): boolean`
Removes buffer at `index` (default: current buffer index). Returns true if the buffer was removed. When you remove the currently selected buffer, the gpu automatically switches back to index 0 (reserved for a screen)

So no argument to `freeBuffer` should be valid. It also seems to still crash OCEmu, it should probably print an error instead. (Even if you might not be able to see it)

I have also discovered that calling `bitblt` without arguments also crashes OCEmu. According to the docs (and my tests in game) that should work fine. Manually passing all the default values does work, however.

**Comment by zenith391** [comment]
Ok I see, `freeBuffer` and `getBufferSize` arguments are now fixed in 574882b
`bitblt` default arguments are fixed too. However about your errors crashing OCEmu, it's not normal. A component error should only cause a crash if the OS doesn't handle it for some reason. But OpenOS does it.
Do you use an alternate OpenComputers OS (MineOS, Fuchas...) ?

**Comment by Xandaros** [comment]
Nope, I'm just using OpenOS. I was using an old copy, but I tried renaming my ocemu.cfg and installing onto a new filesystem with much the same effect.

Though, since it seems to work fine now, I'm not too bothered.

I did try to just call `error` in the Lua REPL, but that gets caught just fine, so it must be components specifically that cause the crash.

Thanks for adding the defaults :)
